iceoryx_hoofs/include/iceoryx_hoofs/internal/units/duration.inl

```diff
@@ -79,31 +84,35 @@ inline constexpr Duration Duration::fromNanoseconds(const T value) noexcept
     auto clampedValue = positiveValueOrClampToZero(value);
     auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::NANOSECS_PER_SEC);
     auto nanoseconds = static_cast<Duration::Nanoseconds_t>(clampedValue % Duration::NANOSECS_PER_SEC);
-    return Duration{seconds, nanoseconds};
+    return createDuration(seconds, nanoseconds);
 }
 template <typename T>
 inline constexpr Duration Duration::fromMicroseconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
-    auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MICROSECS_PER_SEC);
-    auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MICROSECS_PER_SEC)
-                                                            * Duration::NANOSECS_PER_MICROSEC);
-    return Duration{seconds, nanoseconds};
+    const auto clampedValue = positiveValueOrClampToZero(value);
+    const auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MICROSECS_PER_SEC);
+    const auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MICROSECS_PER_SEC)
+                                                                  * Duration::NANOSECS_PER_MICROSEC);
+    return createDuration(seconds, nanoseconds);
 }
 template <typename T>
 inline constexpr Duration Duration::fromMilliseconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
-    auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MILLISECS_PER_SEC);
-    auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MILLISECS_PER_SEC)
-                                                            * Duration::NANOSECS_PER_MILLISEC);
-    return Duration{seconds, nanoseconds};
+    const auto clampedValue = positiveValueOrClampToZero(value);
+    const auto seconds = static_cast<Duration::Seconds_t>(clampedValue / Duration::MILLISECS_PER_SEC);
+    const auto nanoseconds = static_cast<Duration::Nanoseconds_t>((clampedValue % Duration::MILLISECS_PER_SEC)
+                                                                  * Duration::NANOSECS_PER_MILLISEC);
+    return createDuration(seconds, nanoseconds);
 }
 template <typename T>
 inline constexpr Duration Duration::fromSeconds(const T value) noexcept
 {
-    auto clampedValue = positiveValueOrClampToZero(value);
+    const auto clampedValue = positiveValueOrClampToZero(value);
     constexpr Duration::Seconds_t MAX_SECONDS_BEFORE_OVERFLOW{std::numeric_limits<Duration::Seconds_t>::max()};
+
+    // AXIVION Next Construct AutosarC++19_03-M0.1.2, AutosarC++19_03-M0.1.9,
+    // FaultDetection-DeadBranches : False positive, whether this is a dead branch is platforms
```

Comment:
Probably because it cannot happen if bit width of `MAX_SECONDS_BEFORE_OVERFLOW` is of at least as large as `uint64_t`. It is difficult to choose an integer type where this is not the case on any platform that does not have larger integers.

I think the comment is fine for this reason, but ideally there should not be the need of a comment at all...